GoogleApiWrappers/src/Google.Apis.Storage.v1.ClientWrapper/StorageClient.DownloadObject.cs

```diff
@@ -137,49 +150,52 @@ public void DownloadObject(Object source, Stream destination)
             CancellationToken cancellationToken,
             IProgress<IDownloadProgress> progress)
         {
-            string uri = GetUri(source);
-            return DownloadObjectAsyncImpl(uri, destination, options, cancellationToken, progress);
+            var uriBuilder = GetUriBuilder(source);
+            return DownloadObjectAsyncImpl(uriBuilder, destination, options, cancellationToken, progress);
         }
 
         /// <summary>
-        /// Constructs the media URL of an object from its bucket and name.
+        /// Constructs the media URL of an object from its bucket and name. This does not include the generation
+        /// or any preconditions.
         /// </summary>
-        private static string GetUri(string bucket, string objectName)
+        private static StringBuilder GetUriBuilder(string bucket, string objectName)
         {
             ValidateBucket(bucket);
             objectName.CheckNotNull(nameof(objectName));
-            return $"https://www.googleapis.com/download/storage/v1/b/{bucket}/o/{Uri.EscapeDataString(objectName)}?alt=media";
+            return new StringBuilder($"https://www.googleapis.com/download/storage/v1/b/{bucket}/o/{Uri.EscapeDataString(objectName)}?alt=media");
         }
 
         /// <summary>
         /// Obtains the media URL of an object from eiteher the MediaLink property, or its bucket and name.
         /// </summary>
-        private static string GetUri(Object source)
+        private static StringBuilder GetUriBuilder(Object source)
         {
             source.CheckNotNull(nameof(source));
-            if (source.MediaLink != null)
-            {
-                return source.MediaLink;
-            }
             if (source.Bucket != null && source.Name != null)
             {
-                return GetUri(source.Bucket, source.Name);
+                return GetUriBuilder(source.Bucket, source.Name);
             }
-            throw new ArgumentException("Object must have either MediaLink or both Bucket and Name set");
+            throw new ArgumentException("Object must both Bucket and Name set");
         }
 
-        private void DownloadObjectImpl(string uri, Stream destination, DownloadObjectOptions options, IProgress<IDownloadProgress> progress)
+        private void DownloadObjectImpl(
+            StringBuilder uriBuilder,
```

Comment:
I know that we're trying to be efficient here but passing a `StringBuilder` as the way to pass the URI to download seems a strange leak of the implementation details on how the URI is built. Specially because we're passing in an object to be mutated, I think that the method should be the one creating the `StringBuilder` instead, even if that requires an extra string copy.